lib/testutils/src/test_backend.rs

```diff
@@ -0,0 +1,276 @@
+// Copyright 2023 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::Any;
+use std::collections::HashMap;
+use std::fmt::{Debug, Error, Formatter};
+use std::io::{Cursor, Read};
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, Mutex, MutexGuard, OnceLock};
+
+use jj_lib::backend::{
+    make_root_commit, Backend, BackendError, BackendResult, ChangeId, Commit, CommitId, Conflict,
+    ConflictId, FileId, ObjectId, SymlinkId, Tree, TreeId,
+};
+use jj_lib::repo_path::RepoPath;
+
+const HASH_LENGTH: usize = 10;
+const CHANGE_ID_LENGTH: usize = 16;
+
+static BACKEND_DATA: OnceLock<Mutex<HashMap<PathBuf, Arc<Mutex<TestBackendData>>>>> =
+    OnceLock::new();
+
+fn backend_data() -> &'static Mutex<HashMap<PathBuf, Arc<Mutex<TestBackendData>>>> {
+    BACKEND_DATA.get_or_init(|| Mutex::new(HashMap::new()))
+}
+
+#[derive(Default)]
+pub struct TestBackendData {
+    commits: HashMap<CommitId, Commit>,
+    trees: HashMap<RepoPath, HashMap<TreeId, Tree>>,
+    files: HashMap<RepoPath, HashMap<FileId, Vec<u8>>>,
+    symlinks: HashMap<RepoPath, HashMap<SymlinkId, String>>,
+    conflicts: HashMap<RepoPath, HashMap<ConflictId, Conflict>>,
+}
+
+fn test_hash(content: &(impl jj_lib::content_hash::ContentHash + ?Sized)) -> Vec<u8> {
```

Comment:
nit: `test` here sounds like verb, not noun

Reply:
Renamed to `get_hash()`